presto-spi/src/main/java/com/facebook/presto/spi/type/MapType.java

```diff
@@ -99,8 +99,7 @@ public long hash(Block block, int position)
         long result = 0;
 
         for (int i = 0; i < mapBlock.getPositionCount(); i += 2) {
-            result += hashPosition(keyType, mapBlock, i);
```

Comment:
Maybe combine the entries with `^`. Using `+` may cause unintended bias in the bits due to carry.

Reply:
@martint : But we will have the same problem that {a->1, b->2} has the same hash value with {a->2, b->1} right?  (Since we all XOR over all the hash values now.. :( )

Reply:
Or, we can combine the entires with `^`, but compute the entry hash by adding key hash and value hash?

Reply:
I have little knowledge about this. However, even if `+` does have the downside of bias the bit toward the left, `+` still sounds like a better idea than `^` considering the downside of `^` (`x^x=0`)

Reply:
By the way, I noticed that JDK does exactly what you're doing here (sum of xor of hash code). I'm not saying JDK is an example of good hash functions. But I consider it a reasonable minimum.

Reply:
@wenleix, yeah, that's a good point. That could be fixed with some of the tricks murmur and xxhash use (e.g., multiplying by prime numbers and rotating), but don't worry about it.